File: internal/domain/session_context.go
package domain

import (
	"time"
)

// SessionContext represents saved context for a session (simplified schema after migration 002)
type SessionContext struct {
	ID        int
	AgentID   AgentID
	SessionID string
	Context   string    // Markdown-formatted context
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: eye_in_the_sky_web/lib/eye_in_the_sky_web/contexts/session_context.ex
defmodule EyeInTheSkyWeb.Contexts.SessionContext do
  use Ecto.Schema
  import Ecto.Changeset

  schema "session_context" do
    field :context, :string

    belongs_to :agent, EyeInTheSkyWeb.Agents.Agent, type: :string
    # Note: session_id is not a foreign key in the schema, just a field
    field :session_id, :string

    field :created_at, :utc_datetime
    field :updated_at, :utc_datetime
  end

  @doc false
  def changeset(session_context, attrs) do
    session_context
    |> cast(attrs, [:agent_id, :session_id, :context])
    |> validate_required([:agent_id, :session_id, :context])
  end
end
